Added RouteAcquisition File & Downloaded all Data

dodedic opened the pull request:
Iterates over all airports and months of the year, data stored on Sharepoint.
Added .env to .gitignore file.

@michaelweinold for some reason I can't access they key from the .env file, am still working on it.

dodedic committed:
API scripts

dodedic committed:
API RouteAcquisition

dodedic committed:
RouteAcquisition File added

Iterates through all airports and months of the year
Also .gitignore added .env for API key

dodedic commented:
@michaelweinold both older commits 30f20ee and ca8a3c8 contain sensitive data from a week ago, we need to remove those first.

michaelweinold commented:
> we need to remove those first.

Be my guest 😎

michaelweinold commented on ADB/Airport Data/RouteAcquisition.py:
```diff
@@ -0,0 +1,55 @@
+import os
+import requests
+import json
+from dotenv import load_dotenv
```
I would recommend we use native Python functionality instead. [TOML support is relatively new](https://realpython.com/python-toml/). You can take some inspiration from: https://github.com/sustainableaviation/EcoPyLot/blob/8a6b7b5dff270293a38754459cb5f2812471245e/ecopylot/utils.py#L6

michaelweinold commented on ADB/Airport Data/RouteAcquisition.py:
```diff
@@ -0,0 +1,55 @@
+load_dotenv()
+api_key = os.getenv('ADB_KEY')
+
+# Function to retrieve data from the API for a given airport ICAO code and date
```
All fine for a draft PR. When we go to production, I suggest we follow [the NumPy docstrings style guide](https://numpydoc.readthedocs.io/en/latest/format.html).

michaelweinold commented on ADB/Healtcheck.py:
This could later be moved to a single function - with some docstring documentation.

michaelweinold commented:
> I now just simply reset the compromised API key

![hacker-hackerman](https://github.com/sustainableaviation/demandmap/assets/23102087/ac9488f3-4211-44d2-83f6-7825307bcbaa)

michaelweinold commented on ADB/RouteAcquisition - Kopie.py:
Is this file an undesired copy?

michaelweinold reviewed (changes requested):
Nice work - see my comments.

dodedic committed:
Comments implemented

Now using TOML for API Key
added to .gitignore

dodedic commented:
@michaelweinold Changes implemented as per comments.

dodedic committed:
Update .gitignore

dodedic committed:
Update .gitignore

dodedic committed:
Delete ADB/Airport Data/config.toml.txt

Still the old key, was a bit confused and thought the issue was the ignore file, but it was just this old file lingering around

michaelweinold commented:
...some more work needed, @dodedic.

dodedic commented:
I cannot see any new comments though @michaelweinold

michaelweinold reviewed (changes requested):
...can you see my comments now?

dodedic committed:
Comments V2 implemented

Also includes the config.toml template

dodedic committed:
Update .gitignore

Now to ignore the config.toml forever

dodedic commented:
All done! @michaelweinold 
Also added the config.toml file template now as well and changed the .gitignore back to ignore it!

michaelweinold commented on ADB/AirportData/RouteAcquisition.py:
```diff
@@ -0,0 +1,63 @@
+import json
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import toml
```
Excellent - just native imports!

michaelweinold commented on ADB/AirportData/config.toml:
```diff
@@ -0,0 +1,2 @@
+[api]
+key = "API_KEY_HERE"
```
Nice, nice, nice!

michaelweinold commented on ADB/Healtcheck.py:
```diff
@@ -0,0 +1,44 @@
+import toml
+
+
+def get_api_key():
```
As I mentioned earlier (https://github.com/sustainableaviation/demandmap/pull/20#discussion_r1584561530), this is all fine for now. Before we release the code, you will need to add Numpy-style doctrings.
